numba/tests/test_dictimpl.py

```diff
@@ -181,6 +186,80 @@ def __next__(self):
     next = __next__    # needed for py2 only
 
 
+############################################################
+# Supporting code for TestDictImpl.test_parametrized_types
+############################################################
```

Comment:
I suggest putting all this support code in the actual test.

Reply:
Moved everything into the actual test, except for the extension types which need to be module globals to be pickleable/cacheable.